Microsoft.Toolkit.Win32/Microsoft.Toolkit.Forms.UI.XamlHost/WindowsXamlHostBase.Layout.cs

```diff
@@ -45,14 +46,35 @@ public override Size GetPreferredSize(Size proposedSize)
             }
 
             var preferredSize = Size.Empty;
-            if (_xamlSource.Content != null)
+            if ((_xamlSource.Content as DpiScalingPanel).Child != null)
             {
                 preferredSize = new Size((int)_xamlSource.Content.DesiredSize.Width, (int)_xamlSource.Content.DesiredSize.Height);
             }
 
             return preferredSize;
         }
 
+        /// <summary>
+        ///     Sets a scaling factor based on the current dpi value on the scaling panel
+        /// </summary>
+        protected void UpdateDpiScalingFactor()
+        {
+            DpiScalingPanel panel = _xamlSource.Content as DpiScalingPanel;
+            double dpi = 96.0f;
+            if (_xamlIslandWindowHandle != IntPtr.Zero)
+            {
+                uint windowDpi = SafeNativeMethods.GetDpiForWindow(_xamlIslandWindowHandle);
```

Comment:
Has this been tested with various DPI modes and manifest settings? We had a lot of problems with web view that ended up requiring a specific version of .NET Framework and specific settings in the manifest to ensure DPI change events were captured and the values were correct

Reply:
I did some testing with various DPI modes with the scaling property enabled and disabled. There are some winforms specific differences between declaring the dpi mode in app.manifest file or in app.config (for .net 4.7) related to the resizing of the winforms controls but these do not seem to affect the scaling of the XAML content itself. Additionally, recent insider builds apparently now scale the XAML content natively and I have submitted a pull request (#2604) to avoid issues in this case. 

Some of my findings for Windows 10 1809 with DpiScalingRenderTransform=true and false and on the latest insider build (18267, with the pull request mentioned above applied):

**Not dpi aware app**
Windows 1809,  DpiScalingRenderTransform=true or false : Scales correctly (surprisingly)
Insider: Sized correctly, but can be blurry

**System dpi aware app**
Windows 1809,  DpiScalingRenderTransform=false : Size is always at 100% + additional scaling when changing dpi, can be blurry in this case
Windows 1809,  DpiScalingRenderTransform=true:  Scales correctly at initial dpi, but will be blurry when dpi changes dynamically (as the rest of the application)
Insider: same as Windows 1809, DpiScalingRenderTransform=true

**Per monitor (V1) aware app**
Windows 1809,  DpiScalingRenderTransform=false : Size is always 100%, does not scale at all
Windows 1809,  DpiScalingRenderTransform=true:  Scales correctly at initial dpi, but does not rescale when dpi is changed
Insider: same as Windows 1809 with DpiScalingRenderTransform=true

**Per monitor (V2) aware app**
Windows 1809,  DpiScalingRenderTransform=false : Size is always 100%, does not scale at all
Windows 1809,  DpiScalingRenderTransform=true:  Scales correctly
Insider: same as Windows 1809 with DpiScalingRenderTransform=true

The remaining issue seems to be with per monitor (v1) dpi aware apps, but even the native scaling in the recent insider builds does not seem to handle this case correctly.